commons/src/main/java/org/infinispan/commons/configuration/attributes/AttributeSet.java

```diff
@@ -118,10 +121,47 @@ public AttributeSet protect() {
       return protectedSet;
    }
 
+   public boolean isModified() {
+      for(Attribute<?> attribute : attributes.values()) {
+         if (attribute.isModified())
+            return true;
+      }
+      return false;
+   }
+
    public boolean isProtected() {
       return protect;
    }
 
+   public void write(XMLStreamWriter writer, AttributeDefinition<?> def) throws XMLStreamException {
+      write(writer, def, def.name());
+   }
+
+   public void write(XMLStreamWriter writer, AttributeDefinition<?> def, Enum<?> name) throws XMLStreamException {
+      write(writer, def, name.toString());
+   }
+
+   public void write(XMLStreamWriter writer, AttributeDefinition<?> def, String name) throws XMLStreamException {
+      Attribute<?> attribute = attribute(def);
+      attribute.write(writer, name);
+   }
+
+
+   /**
+    * Writes this attributes to the specified XMLStreamWriter
```

Comment:
This Javadoc could be probably removed or we should add proper docs to the rest of the methods (I would vote for the latter).